cleangoldenami/main.go

```diff
@@ -207,13 +260,15 @@ func hasMatchingArchitecture(image ec2Types.Image, architecture ec2Types.Archite
 func filterImages(images []ec2Types.Image) ([]ec2Types.Image, []ec2Types.Image, []ec2Types.Image) {
 	var arm64Images, legacyx86Images, x86Images []ec2Types.Image
 	for _, image := range images {
-
-		if hasVersionTag(image, "rhel-arm64") && hasMatchingArchitecture(image, ArchitectureARM64) {
-			arm64Images = append(arm64Images, image)
-		} else if hasVersionTag(image, "legacy-x86_64") && hasMatchingArchitecture(image, ArchitectureX86) {
-			legacyx86Images = append(legacyx86Images, image)
-		} else if hasVersionTag(image, "rhel-x86_64") && hasMatchingArchitecture(image, ArchitectureX86) {
-			x86Images = append(x86Images, image)
+		versionTag, ok := getImageVersionTag(image)
+		if ok {
+			if versionTag == "rhel-arm64" && hasMatchingArchitecture(image, ArchitectureARM64) {
+				arm64Images = append(arm64Images, image)
+			} else if versionTag == "legacy-x86_64" && hasMatchingArchitecture(image, ArchitectureX86) {
+				legacyx86Images = append(legacyx86Images, image)
+			} else if versionTag == "rhel-x86_64" && hasMatchingArchitecture(image, ArchitectureX86) {
+				x86Images = append(x86Images, image)
+			}
 		}
```

Comment:
do you want `else` (`!ok`) logging here?  Or do you have sufficient trust in `getImageVersionTag` and the tagging on images you might care about?

Reply:
I think it might actually be worth something like `found untagged image %v in region %v!`. I'll add this.